src/MOIWrapper.jl

```diff
@@ -136,30 +170,59 @@ function update_bounds_for_binary_vars!(col_lb::Vector{Float64}, col_ub::Vector{
     end
 end
 
-function update_zeroone_indices(user_optimizer::MOI.ModelLike, mapping::MOIU.IndexMap,
-    ci::Vector{MOI.ConstraintIndex{F,S}}, zero_one_indices::Vector{Int}) where {F, S}
+function update_indices(user_optimizer::MOI.ModelLike, 
+                        mapping::MOIU.IndexMap, 
+                        ci::Vector{MOI.ConstraintIndex{F,S}}, 
+                        list_of_indices::Vector{Int}) where {F, S}
+    
     for i in 1:length(ci)
         f = MOI.get(user_optimizer, MOI.ConstraintFunction(), ci[i])
-        push!(zero_one_indices, mapping.varmap[f.variable].value)
+        push!(list_of_indices, mapping.varmap[f.variable].value)
     end
 end
 
-function update_integer_indices(user_optimizer::MOI.ModelLike, mapping::MOIU.IndexMap,
-    ci::Vector{MOI.ConstraintIndex{F,S}}, integer_indices::Vector{Int}) where {F, S}
-    for i in 1:length(ci)
-        f = MOI.get(user_optimizer, MOI.ConstraintFunction(), ci[i])
-        push!(integer_indices, mapping.varmap[f.variable].value)
+function create_indices_for_bounds(user_optimizer::MOI.ModelLike, mapping::MOIU.IndexMap,
+                                   zeroone_indices::Vector{Int}, integer_indices::Vector{Int})
+    indices = MOI.ConstraintIndex[]
+    num_bounds = 0
+    list_of_constraints = MOI.get(user_optimizer, MOI.ListOfConstraints())
+    for (F,S) in list_of_constraints
+        if F == MOI.SingleVariable
+            list_of_ci = MOI.get(user_optimizer, MOI.ListOfConstraintIndices{F,S}())
+            if S == MOI.ZeroOne
+                update_indices(user_optimizer, mapping, list_of_ci, zeroone_indices)
+            elseif S == MOI.Integer
+                update_indices(user_optimizer, mapping, list_of_ci, integer_indices)
+            end
+            for i in 1:length(list_of_ci)
+                mapping.conmap[list_of_ci[i]] = MOI.ConstraintIndex{F,S}(num_bounds + i)
+            end
+            num_bounds += MOI.get(user_optimizer, MOI.NumberOfConstraints{F,S}())
+        end
     end
 end
 
-"""
-    function copy!(cbc_optimizer, user_optimizer; copynames=false)
+function create_indices_for_rows(cbc_optimizer::CbcOptimizer, user_optimizer::MOI.ModelLike,
```

Comment:
Same thing, please add a comment or docstring explaining what this function does.